k8s/cmd/commands/sidecar/inject_test.go

```diff
@@ -0,0 +1,1057 @@
+
+	//"sigs.k8s.io/kustomize/kyaml/ext"
+
+	//"github.com/stretchr/testify/assert"
+
+	"github.com/v6d-io/v6d/k8s/apis/k8s/v1alpha1"
+	"github.com/v6d-io/v6d/k8s/cmd/commands/flags"
+	"github.com/v6d-io/v6d/k8s/cmd/commands/util"
+	"github.com/v6d-io/v6d/k8s/controllers/k8s"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestInjectCmd(t *testing.T) {
+	testReplicas := struct {
+		name                   string
+		vineyardReplicas       int
+		etcdReplicas           int
+		expectedImage          string
+		expectedCpu            string
+		expectedMemery         string
+		expectedService_port   int
+		expectedService_type   string
+		expectedSolume_pvcname string
+	}{
+		name:                   "test replicas",
+		vineyardReplicas:       1,
+		etcdReplicas:           2,
+		expectedImage:          "vineyardcloudnative/vineyardd:alpine-latest",
+		expectedCpu:            "",
+		expectedMemery:         "",
+		expectedService_port:   9600,
+		expectedService_type:   "ClusterIP",
+		expectedSolume_pvcname: "",
+	}
+	t.Run(testReplicas.name, func(t *testing.T) {
+		// set the flags
+		//flags.KubeConfig = "/tmp/e2e-k8s.config"
+		flags.Namespace = "vineyard-system"
+		flags.KubeConfig = "/home/zhuyi/.kube/config"
+		flags.WorkloadResource = `{
+			"apiVersion": "apps/v1",
+			"kind": "Deployment",
+			"metadata": {
+			  "name": "nginx-deployment",
+			  "namespace": "vineyard-system"
+			},
+			"spec": {
+			  "selector": {
+				"matchLabels": {
+				  "app": "nginx"
+				}
+			  },
+			  "template": {
+				"metadata": {
+				  "labels": {
+					"app": "nginx"
+				  }
+				},
+				"spec": {
+				  "containers": [
+					{
+					  "name": "nginx",
+					  "image": "nginx:1.14.2",
+					  "ports": [
+						{
+						  "containerPort": 80
+						}
+					  ]
+					}
+				  ]
+				}
+			  }
+			}
+		}`
+		flags.VineyarddOpts.Replicas = 1
+		flags.VineyarddOpts.EtcdReplicas = 1
+
+		flags.ApplyResources = true
+		injectCmd.Run(injectCmd, []string{})
+		// get the replicas of etcd and vineyardd
+		k8sclient := util.KubernetesClient()
+		vineyardPods := corev1.PodList{}
+		//vineyardPods := appsv1.DeploymentList{}
+		vineyarddOpts := []client.ListOption{
+			client.InNamespace(flags.Namespace),
+			client.MatchingLabels{
+				"app.vineyard.io/role": "vineyardd",
+			},
+		}
+		err := k8sclient.List(context.Background(), &vineyardPods, vineyarddOpts...)
+		if err != nil {
+			t.Errorf("list vineyardd pods error: %v", err)
+		}
+
+		etcdPod := corev1.PodList{}
+		etcdOpts := []client.ListOption{
+			client.InNamespace(flags.Namespace),
+			client.MatchingLabels{
+				"app.vineyard.io/role": "etcd",
+			},
+		}
+		err = k8sclient.List(context.Background(), &etcdPod, etcdOpts...)
+		if err != nil {
+			t.Errorf("list etcd pods error: %v", err)
+		}
+
+		for _, pod := range vineyardPods.Items {
+			for _, container := range pod.Spec.Containers {
+				if container.Image != testReplicas.expectedImage {
```

Comment:
It's a bug here. Please fix it as the container of the injected workload is not `vineyardcloudnative/vineyardd:alpine-latest`. Also, could you please check the other parts at the same time?